extension/src/LanguageFunctions.ts

```diff
@@ -14,28 +14,28 @@ import { isNullOrUndefined } from 'util';
 
 const logger = Logging.ConsoleLogger.getInstance();
 
-export async function getGXlfDocument() {
-    let gXlfFilePath = await WorkspaceFunctions.getGXlfFile();
-    if (isNullOrUndefined(gXlfFilePath)) {
-        return null;
+export async function getGXlfDocument(): Promise<{ fileName: String; gXlfDoc: Xliff }> {
+
+    let uri = await WorkspaceFunctions.getGXlfFile();
+    if (isNullOrUndefined(uri)) {
+        throw new Error("No g.xlf file was found");
     }
 
-    let gxlfDoc = Xliff.fromFileSync(gXlfFilePath.fsPath, 'UTF8');
-    return gxlfDoc;
+    let gXlfDoc = Xliff.fromFileSync(uri.fsPath, 'UTF8');
+    return { fileName: await VSCodeFunctions.getFilename(uri.fsPath), gXlfDoc: gXlfDoc };
+
 }
 
 export async function updateGXlfFromAlFiles(replaceSelfClosingXlfTags: boolean = true, formatXml: boolean = true) {
-    let gXlfDoc = await getGXlfDocument();
-    if (null === gXlfDoc) {
-        throw new Error("No g.xlf file was found");
-    } else {
-        let alObjects = await WorkspaceFunctions.getAlObjectsFromCurrentWorkspace();
-        alObjects.forEach(alObject => {
-            updateGXlf(gXlfDoc, alObject.getTransUnits());
-        });
-        let gXlfFilePath = await WorkspaceFunctions.getGXlfFile();
-        gXlfDoc.toFileSync(gXlfFilePath.fsPath, replaceSelfClosingXlfTags, formatXml, 'utf8bom');
-    }
+    let gXlfDocument = await getGXlfDocument();
+    let alObjects = await WorkspaceFunctions.getAlObjectsFromCurrentWorkspace();
+    alObjects.forEach(alObject => {
+        updateGXlf(gXlfDocument.gXlfDoc, alObject.getTransUnits());
+    });
+    let gXlfFilePath = await WorkspaceFunctions.getGXlfFile();
+    gXlfDocument.gXlfDoc.toFileSync(gXlfFilePath.fsPath, replaceSelfClosingXlfTags, formatXml, 'utf8bom');
+
+    return gXlfDocument.fileName;
```

Comment:
How does the compiler not say anything about this function not having a return signature?

Reply:
It seems valid. Might be a setting we want to adjust. However in this function we have an expected return type right?

Reply:
It seems to handle implicit return types just fine, with intellisense and all. But I'll add the return type here. 

I guess we have this in a lot of places when you make this mandatory :)

Reply:
Yeah I bet. I'll look in to the options and some point.